### PR TITLE
Add support to use "TryAutoDetect"

### PR DESCRIPTION
**Pull request by xiangyan99**
https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/526

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_enums.py [hunk 1]
```diff
@@ -12,7 +11,7 @@ class RegionalAuthority(str, Enum, metaclass=CaseInsensitiveEnumMeta):
 
     #: Attempt to discover the appropriate authority. This works on some Azure hosts, such as VMs and
     #: Azure Functions. The non-regional authority is used when discovery fails.
-    AUTO_DISCOVER_REGION = ConfidentialClientApplication.ATTEMPT_REGION_DISCOVERY
+    AUTO_DISCOVER_REGION = "tryautodetect"
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_enums.py:14]: Question: It doesn't appear `RegionalAuthority` is exported. Was the intent for users to use the `AUTO_DISCOVER_REGION` when setting `AZURE_REGIONAL_AUTHORITY_NAME`? Or is the typical usage pattern like:

```bash
export AZURE_REGIONAL_AUTHORITY_NAME=true
export AZURE_REGIONAL_AUTHORITY_NAME=TryAutoDetect
```

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # Release History
 
-## 1.13.0b2 (Unreleased)
+## 1.13.0b2 (2023-02-07)
 
 ### Features Added
 
 - Added `AzureDeveloperCredential` for Azure Developer CLI. ([#27916](https://github.com/Azure/azure-sdk-for-python/pull/27916))
+- Added support to use "TryAutoDetect" to enable auto detect.    ([#526](https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/526))
```
Comment (pvaneck) [inline sdk/identity/azure-identity/CHANGELOG.md:8]: Perhaps some additional detail (assuming this usage pattern is correct)?
```suggestion
- Added support to use "TryAutoDetect" as the value for `AZURE_REGIONAL_AUTHORITY_NAME` to enable auto detecting the appropriate authority ([#526](https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/526))
```

File: sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py [hunk 3]
```diff
@@ -31,6 +31,8 @@ def __init__(
         self._instance_discovery = instance_discovery
         self._authority = normalize_authority(authority) if authority else get_default_authority()
         self._regional_authority = os.environ.get(EnvironmentVariables.AZURE_REGIONAL_AUTHORITY_NAME)
+        if self._regional_authority.lower() in ["tryautodetect", "true"]:
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py:34]: Maybe do ` os.environ.get(EnvironmentVariables.AZURE_REGIONAL_AUTHORITY_NAME, "")` above to ensure the `lower()` always works?

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py]: Nvm, better to just check value before lower call as you did in the latest commit.

File: sdk/identity/azure-identity/azure/identity/_enums.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_enums.py:14]: Exposing the consts are not in the scope of this PR. :)

File: sdk/identity/azure-identity/tests/test_certificate_credential.py [hunk 4]
```diff
@@ -170,7 +170,10 @@ def test_regional_authority():
 
         assert mock_confidential_client.call_count == 1
         _, kwargs = mock_confidential_client.call_args
-        assert kwargs["azure_region"] == region.value
+        if region == RegionalAuthority.AUTO_DISCOVER_REGION:
+            assert kwargs["azure_region"]
```
Comment (rayluo) [inline sdk/identity/azure-identity/tests/test_certificate_credential.py:174]: MSAL Python does not guarantee that value will remain a `True`. Better do it this way.

```suggestion
            assert kwargs["azure_region"] == msal.ConfidentialClientApplication.ATTEMPT_REGION_DISCOVERY
```

File: sdk/identity/azure-identity/tests/test_client_secret_credential.py [hunk 5]
```diff
@@ -136,7 +136,10 @@ def test_regional_authority():
 
         assert mock_confidential_client.call_count == 1
         _, kwargs = mock_confidential_client.call_args
-        assert kwargs["azure_region"] == region.value
+        if region == RegionalAuthority.AUTO_DISCOVER_REGION:
+            assert kwargs["azure_region"]
```
Comment (rayluo) [inline sdk/identity/azure-identity/tests/test_client_secret_credential.py:140]: Same suggestion as above.

**Review by pvaneck — APPROVED** [review]

**Pull request merged**